### ISSUE TITLE
EventEmitter memory leak when using with socket.io

### ISSUE THREAD

**Issue by uniibu** [keymetrics/pm2-io-apm#252]
When using `@pm2/io` with socket.io and after a few emits back and forth from the client, we get this warning:
```
43|app | 2019-05-23T23:59:53+00:00: (node:18867) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 16 data listeners added. Use emitter.setMaxListeners() to increase limit
43|app | 2019-05-23T23:59:53+00:00:     at _addListener (events.js:247:17)
43|app | 2019-05-23T23:59:53+00:00:     at Socket.addListener (events.js:263:10)
43|app | 2019-05-23T23:59:53+00:00:     at Socket.Readable.on (_stream_readable.js:822:35)
43|app | 2019-05-23T23:59:53+00:00:     at Socket.<anonymous> (/home/user/.nvm/versions/node/v10.15.3/lib/node_modules/pm2/node_modules/@pm2/io/build/main/metrics/network.js:78:26)
43|app | 2019-05-23T23:59:53+00:00:     at maybeReadMore_ (_stream_readable.js:594:12)
43|app | 2019-05-23T23:59:53+00:00:     at process._tickCallback (internal/process/next_tick.js:63:19)
```
And this was due to `monkeypatching` the `net.Socket.prototype.read`
https://github.com/keymetrics/pm2-io-apm/blob/55b8f1a613931079a0965b5ef95ef50a926a448b/src/metrics/network.ts#L90-L94

**Comment by Eywek** [comment]
Hi,
Can you try with 4.2.2?

**Comment by Unitech** [comment]
@shimiml4 validated that this fixed the leak with EE

**Comment by lkho** [comment]
https://github.com/keymetrics/pm2-io-apm/issues/257

seems this is still not fixed in 4.3.5?
